### PR TITLE
Fixes bug #41214

### LINKED ISSUE #41214: On AWS EC2: salt-cloud create -f create_keypair  does not returns the private key
In order to create a new keypair on AWS EC2 you have to run the corresponding salt-cloud command and in theory you should get back the the private key pair (along with the other response info) but it does not work 

```bash
salt-cloud create -f create_keypair ec2 keyname=mykeyname
```
Using salt-cloud 2016.11.4 (Carbon) it does not shows the private key pair nor relevant information.

### PR DESCRIPTION
**Pull request by colegatron**
Fixes #41214: On AWS EC2: salt-cloud create -f create_keypair does not returns the private key

### What does this PR do?
Now it returns the private key

### What issues does this PR fix or reference?
None

### Tests written?
No

### Commits signed with GPG?
No

### CODE REVIEW

**Comment by cachedout** [comment]
@gtmanfred Do you want to take a look here?

**Review by gtmanfred — APPROVED** [review]

File: salt/cloud/clouds/ec2.py [hunk 1]
```diff
@@ -4136,11 +4136,14 @@ def create_keypair(kwargs=None, call=None):
 
     data = aws.query(params,
                      return_url=True,
+                     return_root=True,
                      location=get_location(),
                      provider=get_provider(),
                      opts=__opts__,
                      sigver='4')
-    return data
+    keys = [ x for x in data[0] if 'requestId' not in x ]
```
Comment (gtmanfred) [inline salt/cloud/clouds/ec2.py]: can we do a check to make sure that we get the correct https status code from the query before we try to parse the data?

Also, this is going to have some lint warnings, you will need to remove the extra spaces between `[x` and `x]`

File: salt/cloud/clouds/ec2.py [hunk 2]
```diff
@@ -4136,11 +4136,14 @@ def create_keypair(kwargs=None, call=None):
...
+    
+    return ( keys, data[1] )
```
Comment (gtmanfred) [inline salt/cloud/clouds/ec2.py]: extra spaces here too for lint.

**Review by gtmanfred — CHANGES_REQUESTED** [review]

File: salt/cloud/clouds/ec2.py (hunk 1 above)
Comment (colegatron) [reply salt/cloud/clouds/ec2.py]: I am sorry. This is something I did a year ago. don't have time to re-setup the env to fix and push again. 
feel free to update yourself, don't worry about the pr credit

Comment (gtmanfred) [reply salt/cloud/clouds/ec2.py]: Ok, i will just push them to your repo once I get a chance to make the changes

**Comment by cachedout** [comment]
@gtmanfred I fixed the lint errors.

**Comment by gtmanfred** [comment]
Ok, cool, i will fix my other issues with it later, or if they ever actually pop up.  It should only be a problem if amazon's api throws an error.

**Review by gtmanfred — APPROVED** [review]

**Comment by TheLastProject** [comment]
@gtmanfred Isn't https://github.com/colegatron/salt/pull/2/commits/d7eaa1a1eda67f79e04d067a4cc33dc044c2676f all that needs to happen?

**Comment by gtmanfred** [comment]
maybe, i haven't had time to look.

If that fixes it, once this is in, could you open a new pr with that change?

**Pull request merged**